### PR TITLE
Start on SequenceCollections refactor

### PR DESCRIPTION
**Pull request by KatherineCaley**
working towards #1867

questions wrt direction are commented in place

### CODE REVIEW

File: src/cogent3/core/new_alignment.py [hunk 1]
```diff
@@ -199,7 +208,10 @@ def to_rich_dict(self):
         else:
             start, stop = self.start, self.stop
```
Comment (KatherineCaley) [inline src/cogent3/core/new_alignment.py:210]: line 213 changes `.seq` from a `SeqData` instance, to a `str`. My instinct is that it should be a new `SeqData` instance where the seq corresponding to `seqid` is sliced

File: src/cogent3/core/new_alignment.py [hunk 2]
```diff
@@ -357,6 +369,101 @@ def _(seq: numpy.ndarray, moltype: MolType) -> tuple[str, IndelMap]:
...
+    label_to_name: Callable = None,
+    info: dict = None,
+    source: Union[str, Path] = None,
+    **kwargs,
+):
+    """Initialize an unaligned collection of sequences.
+
+    Parameters
+    ----------
+    data
```
Comment (KatherineCaley) [inline src/cogent3/core/new_alignment.py:422]: let's decide what forms of `data` we are going to support being passed to `make_unaligned_seqs`. currently I am supporting: 
- `dict{name: seq, ...}`
- `[[name, seq]...]` (lists or sets) 
- `[seq, ...]` (lists or sets) 
- `2D numpy array`
which of these are actually necessary and which have I missed?

File: src/cogent3/core/new_alignment.py [hunk 3]
```diff
@@ -357,6 +369,101 @@ def _(seq: numpy.ndarray, moltype: MolType) -> tuple[str, IndelMap]:
...
+    **kwargs
+        other keyword arguments to be passed to SequenceCollection
+    """
+    moltype = get_moltype(moltype)
+    alphabet = moltype.alphabet
+
+    if len(data) == 0:
+        raise ValueError("data must be at least one sequence.")
+    prepped_data = prep_for_seq_data(data, alphabet)
+
```
Comment (KatherineCaley) [inline src/cogent3/core/new_alignment.py:443]: naming consistency (using `validate_names`) is currently the responsibility of `SeqData` but I think it should be moved into the `SequenceCollection`

File: src/cogent3/core/new_alignment.py (hunk 2 above)
Comment (KatherineCaley) [reply src/cogent3/core/new_alignment.py:422]: Should we support creating a `SequenceCollection` where data is already a `SeqData` instance?

File: tests/test_core/test_new_alignment.py [hunk 4]
```diff
@@ -9,6 +9,9 @@
     AlignedDataView,
     SeqData,
     SeqDataView,
+    SequenceCollection,
+    _names_list_tuple,
+    make_unaligned_seqs,
     seq_index,
```
Comment (GavinHuttley) [inline tests/test_core/test_new_alignment.py:15]: Change all imports to use the new versions of the MolType sequence and alphabet modules.

File: src/cogent3/core/new_alignment.py [hunk 5]
```diff
@@ -357,6 +369,101 @@ def _(seq: numpy.ndarray, moltype: MolType) -> tuple[str, IndelMap]:
...
+    seq_data = SeqData(data=prepped_data, alphabet=alphabet)
+
+    return SequenceCollection(seq_data=seq_data, moltype=moltype)
+
+
+class SequenceCollection:
+    def __init__(self, seq_data: SeqData, moltype: MolType):
```
Comment (GavinHuttley) [inline src/cogent3/core/new_alignment.py:450]: We should require these to be keyword arguments. So `(self, *, ...)`

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
Great progress!
I'm going to get all radical here and suggest that we rename `SeqData` to `SeqsData` because it can contain many sequences.
I also need to update the testing code so that it uses the new versions that I just pushed in for molecular type, alphabets and sequence.

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
Great progress, and some good issues raised. A few small comments to address small comments to address.

File: src/cogent3/core/new_alignment.py [hunk 6]
```diff
@@ -222,49 +172,40 @@ def from_rich_dict(cls, data: dict):
...
-                seq=self.seq,
-                start=self.start,
-                stop=self.stop,
-                step=self.step,
-                offset=self.offset,
-                seqid=self.seqid,
-                seq_len=self.seq_len,
-            )
-        return self.from_rich_dict(self.to_rich_dict())
+        return self.__class__(
```
[the remark this answers is not in the document]
Comment (KatherineCaley) [reply src/cogent3/core/new_alignment.py:151]: @GavinHuttley can you dumb it down to me what you mean by "just return self" - is it in the context of supporting sliced, or to replace the current code?

[the remark this answers is not in the document]
Comment (GavinHuttley) [reply src/cogent3/core/new_alignment.py:151]: Sorry, too brief a comment! I meant the following
```python
def copy(self):
    return self
```
